core/lib/vm/src/oracles/storage.rs

```diff
@@ -313,6 +337,12 @@ impl<S: WriteStorage, H: HistoryMode> VmStorageOracle for StorageOracle<S, H> {
             let to_pay_by_user = self.base_price_for_write_query(&query);
             let prepaid = self.prepaid_for_write(&storage_key);
 
+            let initial_value = self
```

Comment:
Inside `self.write_value(query` the initial value is also updated, please remove it from here